packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -155,6 +155,80 @@ abstract class XcodeBasedProject extends FlutterProjectPlatform  {
     return _projectInfo ??= await xcodeProjectInterpreter.getInfo(hostAppRoot.path);
   }
   XcodeProjectInfo? _projectInfo;
+
```

Comment:
None of this code is new. It was moved from `IosProject` to the base class `XcodeBasedProject` so it can be used by the subclass `MacOSProject`.